src/main/java/javax/jmdns/ServiceInfo.java

```diff
@@ -436,11 +436,19 @@ public static ServiceInfo create(final Map<Fields, String> qualifiedNameMap, fin
 
     /**
      * Get the name of the server.
+     * <b>Note:</b> This will be an empty string if no value has been set
```

Comment:
The `_server` field is initialised in the constructor with the `text` parameter, so there should not be a such situation as "no value has been set` - and if there is: Maybe the right place to address it would be the constructor?

Reply:
I may need to take another look at this. It would seem that the value of the `text` is itself an empty string or not valid input into the `ByteWrangler.encodeText()` method. I know that in my tests by the time the service was resolved the value of `_server == null`